istioctl/cmd/internal-debug.go

```diff
@@ -83,36 +145,31 @@ By default it will use the default serviceAccount from (istio-system) namespace
 				}
 			}
 			var xdsRequest xdsapi.DiscoveryRequest
-			var namespace, serviceAccount string
-			if len(args) > 1 {
-				podName, ns, err := handlers.InferPodInfoFromTypedResource(args[1],
-					handlers.HandleNamespace(namespace, defaultNamespace),
-					kubeClient.UtilFactory())
-				if err != nil {
-					return err
-				}
-				pod, err := kubeClient.CoreV1().Pods(ns).Get(context.TODO(), podName, metav1.GetOptions{})
-				if err != nil {
-					return err
+			var namespace, serviceAccount, resourceName, cmdName string
+			if strings.Contains(args[0], "?proxyID=") {
```

Comment:
proxyID != a pod name (try `edsz?proxyID=i` and see :slightly_smiling_face: ). I am not sure why we need any preprocessing at all?

Reply:
@howardjohn here is the output:
`$ istioctl x internal-debug edsz?proxyID=i`
`Error: pods "i" not found`
As you can see this error information,  we can not control the parameters that user input, so we need to do some pre validation. Besides the proxyID parameter, I also check the sub command of internale-debug. You know, user may execute command like this `istioctl x internal-debug edszx`, however, it's not supported for `edszx`, but `edsz`.

Reply:
BTW,  there should be another feature for parameter of proxyID. It can be used for filtering particular proxyID information, but it's not fully completed for now to many debug APIs. 
Of course, I will continue to implement the multiple istiod pods' aggregation for debug commands after you approve this PR. ^_^